compiler/ast/ast_types.nim

```diff
@@ -1595,10 +1594,15 @@ type
   TSym* {.acyclic.} = object of TIdObj # Keep in sync with PackedSym
     ## proc and type instantiations are cached in the generic symbol
     case kind*: TSymKind
-    of routineKinds:
+    of routineKinds - {skMacro}:
       #procInstCache*: seq[PInstantiation]
       gcUnsafetyReason*: PSym  ## for better error messages regarding gcsafe
       transformedBody*: PNode  ## cached body after transf pass
+    of skMacro:
```

Comment:
I really like this a lot -- it moves macros much closer to "just" a function/procedure.

I can easily imagine building upon this incrementlaly in two very interesting ways:
1. return types of macros as "typed code values" (we know this is an `int` expression or `noreturn` stmt)
2. analysing macros for effects: does it only observe ast, does it change ast, does it read/write macrocache, etc
These could then be combined to determine whether we can cache macro results, trust a macro's output in sem, defer macro evaluation until generic instantiation, and more

_very cool_